From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Raul Munoz <raulrm@qti.qualcomm.com>
Date: Tue, 12 Aug 2026 00:00:00 +0000
Subject: [PATCH] arm64: dts: qcom: monaco-evk: Enable USB host mode on usb_2

The Monaco EVK has a micro-USB OTG port (usb_2 / a400000, labelled USB0)
whose dr_mode is left unset in monaco-evk-common.dtsi.  The
gpio-usb-b-connector driver reads the ID GPIO and, because no OTG cable
grounds the ID pin, defaults the port to peripheral mode and never asserts
VBUS.  USB devices plugged into USB0 are therefore invisible to the kernel.

Set dr_mode = "host" on usb_2 and mark the usb2_vbus fixed regulator
always-on so the 5 V VBUS rail is asserted at boot.  This is needed to
enumerate USB cameras (e.g. EMEET C960) on the IQ-8275 EVK.

Signed-off-by: Raul Munoz <raulrm@qti.qualcomm.com>
---
 arch/arm64/boot/dts/qcom/monaco-evk-common.dtsi | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/qcom/monaco-evk-common.dtsi b/arch/arm64/boot/dts/qcom/monaco-evk-common.dtsi
--- a/arch/arm64/boot/dts/qcom/monaco-evk-common.dtsi
+++ b/arch/arm64/boot/dts/qcom/monaco-evk-common.dtsi
@@ -81,6 +81,7 @@
 	usb2_vbus: regulator-usb2-vbus {
 		compatible = "regulator-fixed";
 		regulator-name = "usb2_vbus";
+		regulator-always-on;
 		gpio = <&pmm8650au_1_gpios 7 GPIO_ACTIVE_HIGH>;
 		regulator-min-microvolt = <5000000>;
 		regulator-max-microvolt = <5000000>;
@@ -927,6 +928,7 @@
 &usb_2 {
+	dr_mode = "host";
 	status = "okay";
 };
